### DISCUSSION TITLE
Using a model trained with nnunet for predicting without using nnunet

### DISCUSSION THREAD

**Discussion by arseneaultj** [MIC-DKFZ/nnUNet#773] [Q&A] [answered]
Hi everyone,

Is there a way to use a model trained with nnunet for predicting without using nnunet? Basically applying the weights on a model with the same architecture using Pytorch?

If yes, is there any preprocessing that needs to be done before the predictions?

**Comment by FabianIsensee** [comment, marked as answer]
You need to do the same preprocessing as nnU-Net does. Exactly. So it's best to use nnU-Net code for that.
If you follow the inference pipeline you can see how the model is restored from a checkpoint. You can then export it however you like, for example as onnx
Best,
Fabian
